### PR TITLE
Feature/support pause

### PR DESCRIPTION
**Pull request by neophob**
enhance ui:
- support game pause
- add option to load specific game part
- stop current game
- load and save current game
- update game log level

### CODE REVIEW

File: anotherworld/video.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func (video *Video) PlaySound(resNum, freq, vol, channel int) {
 		return
 	}
 
-	logger.Info("playSound resNum %d", resNum)
```
Comment (neophob) [inline anotherworld/video.go:276]: less spam

File: wasm/js-canvas.go [hunk 2]
```diff
@@ -20,26 +19,10 @@ func GetCanvas(domElementId string) Canvas {
...
-}
-
-func (c Canvas) fillStyle(style string) {
-	c.context2d.Set("fillStyle", style)
-}
-
-func (c Canvas) fillRect(x, y, width, height int) {
-	c.context2d.Call("fillRect", x, y, width, height)
-}
-
```
Comment (neophob) [inline wasm/js-canvas.go:38]: red diff is a good diff!

File: wasm/index.html [hunk 3]
```diff
@@ -32,6 +42,30 @@
...
+      <a id="badge" href="?gamePart=3">3</a>
+      <a id="badge" href="?gamePart=4">4</a>
+      <a id="badge" href="?gamePart=5">5</a>
+      <a id="badge" href="?gamePart=6">6</a>
+      <a id="badge" href="?gamePart=7">7</a>
+
+      //
+      <a id="badge" href="#" onclick="shutdown()">STOP</a>
+      <a id="badge" href="#" onclick="handleKeyEvent('l', 76, 'keydown'); setTimeout(() => handleKeyEvent('l', 76, 'keyup'), 50);">LOAD</a>
+      <a id="badge" href="#" onclick="handleKeyEvent('s', 83, 'keydown'); setTimeout(() => handleKeyEvent('s', 83, 'keyup'), 50);">SAVE</a>
```
Comment (neophob) [inline wasm/index.html:59]: see https://github.com/neophob/ganother-world/issues/45

File: wasm/index.html [hunk 4]
```diff
@@ -32,6 +42,29 @@
     <h1>
       <img src="logo.png" alt="Gother World">
     </h1>
+    <p>
+      <a id="badge" href="#" onclick="shutdown()">STOP</a>
+      <a id="badge" href="#" onclick="handleKeyEvent('l', 76, 'keydown'); setTimeout(() => handleKeyEvent('l', 76, 'keyup'), 50);">LOAD</a>
+      <a id="badge" href="#" onclick="handleKeyEvent('s', 83, 'keydown'); setTimeout(() => handleKeyEvent('s', 83, 'keyup'), 50);">SAVE</a>
+
+      // Game Part
+      <a id="badge" href="?gamePart=0">0</a>
```
Comment (pfiaux) [inline wasm/index.html:51]: Should badge be a class and not an id?

File: wasm/engine.go [hunk 5]
```diff
@@ -38,6 +38,10 @@ func (engine *Engine) initGame(memlist []byte, bankFilesMap map[int][]byte) {
 }
 
 func (engine *Engine) mainLoop() {
+	if engine.keyMap[anotherworld.KeyPause] == true {
```
Comment (pfiaux) [inline wasm/engine.go:41]: Should pause be a "hold" or a state in the engine that we toggle? Press to pause press again to unpause? feels more common.

Comment (pfiaux) [reply wasm/engine.go:41]: Also if we had an engine state wouldn't that allow a more efficent pause state? the hal could check and slow down the blit while paused (no point rerendering the same frame over and over no?)

File: wasm/index.html (hunk 4 above)
Comment (neophob) [reply wasm/index.html:51]: no, class is evaluated before css and would not work while id is evaluated later. tbh - I'm not the css master, but what I know is class does not work ;)

File: wasm/engine.go (hunk 5 above)
Comment (neophob) [reply wasm/engine.go:41]: My goal is to solve this with #45

Comment (neophob) [reply wasm/engine.go:41]: yeah.. used id because i know shit about css ;)

Comment (neophob) [reply wasm/engine.go:41]: fixed btw

File: wasm/index.html (hunk 4 above)
Comment (neophob) [reply wasm/index.html]: yeah.. used id because i know shit about css ;)

fixed btw

File: wasm/index.html [hunk 6]
```diff
@@ -32,6 +42,29 @@
     <h1>
       <img src="logo.png" alt="Gother World">
     </h1>
+    <p>
+      <a class="badge" href="#" onclick="shutdown()">STOP</a>
+      <a class="badge" href="#" onclick="handleKeyEvent('l', 76, 'keydown'); setTimeout(() => handleKeyEvent('l', 76, 'keyup'), 50);">LOAD</a>
+      <a class="badge" href="#" onclick="handleKeyEvent('s', 83, 'keydown'); setTimeout(() => handleKeyEvent('s', 83, 'keyup'), 50);">SAVE</a>
```
Comment (pfiaux) [inline wasm/index.html:48]: minor magic numbers used for the timeout

Comment (neophob) [reply wasm/index.html:48]: yeah should be fixed with #45 as well

**Pull request merged**